### PR TITLE
Add user to the header using useUser hook

### PR DESCRIPTION
**Pull request by NicholasBottone**
A demonstration of the `useUser` hook that adds the user profile information to the top right of the header navbar. Also adds links to profile page, login, and logout on the header.

### CODE REVIEW

File: components/Header.tsx [hunk 1]
```diff
@@ -1,26 +1,67 @@
...
+                <NavDropdown
+                  title={
+                    <span>
+                      {user.name}{" "}
+                      <Image
+                        src={user.profilePicture}
+                        alt="Profile picture"
+                        width="30"
+                        height="30"
+                        roundedCircle
```
Comment (tkato1) [inline components/Header.tsx:50]: nice use of roundedCircle for a more aesthetic interface!

File: lib/hooks/auth.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 import IUser from "../../types/IUser";
 
 const fetcher = (url: string) => fetch(url).then((r) => r.json());
 
 /**
  * Gets the user that is currently logged in.
  */
-export const useUser = useSWR<IUser>("/api/auth", fetcher);
+export const useUser = () => {
+  const { data, error } = useSWR<IUser>("/api/auth", fetcher);
```
Comment (tkato1) [inline lib/hooks/auth.ts:10]: Nice use of separating difference uses into separate files.

**Review by tkato1 — COMMENTED** [review]
Nice!

File: components/Header.tsx [hunk 3]
```diff
@@ -1,26 +1,67 @@
 import React from "react";
-import { Navbar, Container, Nav, Image } from "react-bootstrap";
+import Link from "next/link";
+import { Navbar, Container, Nav, Image, NavDropdown } from "react-bootstrap";
+import { login, logout, useUser } from "../lib/hooks/auth";
 
 export default function Header() {
+  const { user } = useUser();
```
Comment (christinej604) [inline components/Header.tsx:7]: thank you! - user profile team

File: components/Header.tsx (hunk 1 above)
Comment (christinej604) [inline components/Header.tsx:50]: we needed it to be a rounded circle. that is perfect!

File: components/Header.tsx [hunk 4]
```diff
@@ -1,26 +1,67 @@
...
+                        width="30"
+                        height="30"
+                        roundedCircle
+                      />
+                    </span>
+                  }
+                >
+                  <Link href="/profile">
+                    <NavDropdown.Item as="a" href="/profile">
+                      Profile
```
Comment (christinej604) [inline components/Header.tsx:57]: will update this link after merging

**Review by christinej604 — APPROVED** [review]

**Pull request merged**